library/std/src/sync/mutex/tests.rs

```diff
@@ -236,3 +236,40 @@ fn test_mutex_unsized() {
     let comp: &[i32] = &[4, 2, 5];
     assert_eq!(&*mutex.lock().unwrap(), comp);
 }
+
+#[bench]
+fn bench_lock_uncontended(b: &mut test::Bencher) {
+    let mutex = Mutex::new(0u64);
+
+    b.iter(|| {
+        for _i in 0..10_000 {
+            let mut guard = mutex.lock().unwrap();
+            *guard += 1;
+        }
+    });
+}
+
+#[bench]
+fn bench_lock_contended(b: &mut test::Bencher) {
+    b.iter(|| {
+        let mutex = Arc::new(Mutex::new(0));
+        let mutex2 = Arc::clone(&mutex);
+        thread::spawn(move || {
```

Comment:
This spawns a thread inside the benchmark loop. This is likely messing up benchmarks.

`b.iter` already does aggregation of multiple calls, so if possible you should only measure the operation of interest inside `b.iter` and do the setup outside.